dags/top-downloads-from-sowflake.py

```diff
@@ -92,6 +96,64 @@
 LIMIT 10;
 """
 
+QUERY_WITHOUT_LIMIT = """
+WITH PUBLIC_PROJECTS AS (
+    SELECT
+        node_latest.project_id,
+        node_latest.name
+    FROM
+        synapse_data_warehouse.synapse.node_latest
+    WHERE
+        node_latest.is_public AND
+        node_latest.node_type = 'project'
+),
+DEDUP_FILEHANDLE AS (
+    SELECT DISTINCT
+        PUBLIC_PROJECTS.name,
+        filedownload.user_id,
+        filedownload.file_handle_id AS FD_FILE_HANDLE_ID,
+        filedownload.record_date,
+        filedownload.project_id,
+        file_latest.content_size
+    FROM
+        synapse_data_warehouse.synapse.filedownload
+    INNER JOIN
+        PUBLIC_PROJECTS
+    ON
+        filedownload.project_id = PUBLIC_PROJECTS.project_id
+    INNER JOIN
+        synapse_data_warehouse.synapse.file_latest
+    ON
+        filedownload.file_handle_id = file_latest.id
+    WHERE
+        filedownload.record_date = DATEADD(HOUR, -24, CURRENT_DATE)
```

Comment:
Just thinking ahead, but if the DAG ever fails or if Airflow goes down, do we want a "more robust" way to backfill information?

Reply:
Good idea - I added in a time delta parameter that can be passed into the DAG with 24 as the default. This allows us to back-fill previous days.

![image](https://github.com/Sage-Bionetworks-Workflows/orca-recipes/assets/17128019/b1969162-39ec-4618-ba47-fb1a9b898f49)

Since this would also post to slack (Maybe we want to disable in the future) I updated the title of the message to be either `Yesterday` or the specific date if > 24 hours time delta.
```
Top Downloaded Public Synapse Projects Yesterday!
Top Downloaded Public Synapse Projects 2024-03-02!
```